### ISSUE TITLE
3-d Label Annotation When Painting Outside Z-Axis Range Crashes the Viewer

### ISSUE THREAD

**Issue by Karl5766** [napari/napari#7325]
### 🐛 Bug Report

Using Napari's labels layer to annotate a 3D dataset, when the camera's z position is above the labels layer's maximum z value, the paint brush and eraser tools crush the viewer on paint.

### 💡 Steps to Reproduce

1. `script.py`
```
import napari
import numpy as np

im = np.ones((40,) * 3, dtype=np.int32)
mask = np.ones((20,) * 3, dtype=np.int32)

viewer = napari.Viewer(ndisplay=2)
viewer.add_labels(im, name='image')
viewer.add_labels(mask, name='mask')
viewer.show(block=True)
```
2. In `mask` layer, use paint brush tool, move camera to a location where `z >= 20`
3. paint "above" the mask array

I get the error `Index Error: index 20 is out of bounds for axis 0 with size 20` after step 3) if drawing within the xy limits of the mask layer, or `ValueError: Data is too large` if painting outside all xyz limits of the mask. 

On my PyCharm IDE this throws a bug in the terminal; however, on another Windows computer we have, this crushes the viewer; I'm curious why this is the case?

### 💡 Expected Behavior

No crush or error thrown when this happens.

### 🌎 Environment

napari: napari-0.5.4rc4.dev5+g4157d6cf (latest main branch)
Platform: Windows 11
Python: 3.10
numpy: 2.1.2 (latest)

### 💡 Additional Context

In the above setup, the brush will not crush if it is set to `nedit=3` and if the paint brush stroke does not change any pixel color; however, if the paint location is above the maximum z limit and the mask volume have some pixels with a different color (why?), then it will throw the same errors as when `nedit=2`. 
Same thing happens with 2D/3D eraser.

**Comment by Czaki** [comment]
It looks like we should add to begin of `paint` method:

https://github.com/napari/napari/blob/4157d6cf0cb953782f7af734663c56e830695b18/napari/layers/labels/labels.py#L1191-L1206

same guard like for `fill` method 

https://github.com/napari/napari/blob/4157d6cf0cb953782f7af734663c56e830695b18/napari/layers/labels/labels.py#L1107-L1110

**Comment by dalthviz** [comment]
Hi there, I gave it a check to the proposed solution locally and seems like that prevents raising the error :+1: Could it make sense if I submit a PR that adds a simple test for the `paint` method and the proposed fix?

**Comment by DragaDoncila** [comment]
@dalthviz would be great to see that fix and a test in a PR!

**Comment by jni** [comment]
Wait wait, that's actually a regression — in the image plane you actually want to be able to paint outside the border so you can do some "fine trimming" around the edges of the image. Demo:

https://github.com/user-attachments/assets/ff652c52-e838-4000-b49e-d49a667793da

This is not used in fill because there is no such thing as "brush size" in fill.

So the check needs to be a bit more clever than that — something about not visible dimensions vs visible ones...? 🤔

**Comment by jni** [comment]
It's not clear whether one wants the same behaviour with out-of-slice painting. ie if I paint 10 slices away from the data with a radius 20 brush, do I make a partial sphere in the data off-screen? My instinct is no but I'm not sure (you could say that violates dimensional consistency or something 😂 but I think it's correct to say displayed dimensions are fundamentally different from non-displayed ones.)

**Comment by DragaDoncila** [comment]
>  in the image plane you actually want to be able to paint outside the border so you can do some "fine trimming" around the edges of the image

Ohh yeah that's true, thanks for catching that!

**Comment by brisvag** [comment]
This is marked as high prioroty, but bumping the milestone every time we release won't realistically make us faster at fixing this. Removing milestone until we actually find a way to tackle it in a PR.

**Comment by psobolewskiPhD** [comment]
I can't reproduce this -- it's not super clear from the OP what the steps are (2d? 3d? what n_edit_dim?) but I tried a buncha things and could not get anything unexpected.
Maybe this was yet another issue fixed by https://github.com/napari/napari/pull/8636 ?
@jni were you able to reproduce before?
